### PR TITLE
Soybean and rice plants, tofu, tofu bread recipe

### PR DESCRIPTION
**Pull request by lapatison**
## About the PR

**Media**

Screenshot_1
Screenshot_2

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- add: Added soybeans (juiceable into soy milk) and rice, along with soy sauce and tofu and tofu bread recipes.
- tweak: Adjusted soy milk and oat milk colors

### CODE REVIEW

File: Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml [hunk 1]
```diff
@@ -467,36 +467,3 @@
...
-  - type: SolutionContainerManager
-    solutions:
-      drink:
-        maxVol: 100
-        reagents:
-        - ReagentId: MilkSoy
-          Quantity: 100
-  - type: Drink
-  - type: Sprite
-    sprite: Objects/Consumable/Drinks/soymilk.rsi
```
Comment (Torun-S) [inline Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml:502]: Why remove this entirely? The commit says duplicate milk bag, but I can't find the alleged other one.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml [hunk 2]
```diff
@@ -249,10 +249,10 @@
   - type: SolutionContainerManager
     solutions:
       food:
-        maxVol: 20
+        maxVol: 50
         reagents:
         - ReagentId: MilkOat
-          Quantity: 20
+          Quantity: 50
```
Comment (Torun-S) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml:255]: Good catch.

File: Resources/Prototypes/Recipes/Reactions/food.yml [hunk 3]
```diff
@@ -87,6 +87,19 @@
...
 
+- type: reaction
+  id: CreateTofu
+  impact: Low
+  quantized: true
+  reactants:
+    MilkSoy:
+      amount: 10
+    Enzyme:
+      amount: 5
```
Comment (Torun-S) [inline Resources/Prototypes/Recipes/Reactions/food.yml:98]: Seems to me that from this and the recipes, you intend tofu to just be one item, but on my fork, I was just preparing Tofu as well and intended to make it be made in bigger blocks and sliced for recipes (or to snack on). You might consider doing that, and if you need the sprites for it,

I can arrange PRing it to your branch as I had a community member make some very fine ones for that. I understand it'd mean re-doing a few things, though: Recipe, the item prototype, adding another one, altering the food recipes...

**Review by Torun-S — COMMENTED** [review]
I just so happen to be changing a bunch of cooking things downstream (https://github.com/Nyanotrasen/Nyanotrasen/pull/1255) so this piqued my interest since I was going to work on Tofu and eventually rice. I was going to do it a bit differently, however, see comments.

File: Resources/Prototypes/Recipes/Reactions/food.yml (hunk 3 above)
Comment (Torun-S) [reply Resources/Prototypes/Recipes/Reactions/food.yml:98]: image
image
Preview of the sprites I will use downstream, courtesy of rosysyntax#6514

File: Resources/Prototypes/Hydroponics/seeds.yml [hunk 4]
```diff
@@ -716,3 +716,50 @@
...
+  production: 4
+  yield: 3
+  potency: 5
+  idealLight: 7
+  nutrientConsumption: 0.15
+  chemicals:
+    Nutriment:
+      Min: 1
+      Max: 5
+      PotencyDivisor: 20
```
Comment (Torun-S) [inline Resources/Prototypes/Hydroponics/seeds.yml:765]: Also, isn't the soybean seed prototype missing the soy milk reagent in the chemicals component? Would it then not produce any like this?

Comment (lapatison) [reply Resources/Prototypes/Hydroponics/seeds.yml:765]: Here it is
https://github.com/space-wizards/space-station-14/blob/b474b391c0736f885e3a69fd193de6755478cd98/Resources/Prototypes/Entities/Objects/Consumable/Food/produce.yml#L920-L924

File: Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml (hunk 1 above)
Comment (lapatison) [reply Resources/Prototypes/Entities/Objects/Consumable/Drinks/drinks_bottles.yml:502]: These entities are basically duplicates of these two entities (which are used everywhere):
https://github.com/space-wizards/space-station-14/blob/b474b391c0736f885e3a69fd193de6755478cd98/Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml#L209-L239
But now I intend to revert this removal, as it causes map-related test error and I don't want to break something.

File: Resources/Prototypes/Recipes/Reactions/food.yml (hunk 3 above)
Comment (lapatison) [reply Resources/Prototypes/Recipes/Reactions/food.yml:98]: > I can arrange PRing it to your branch

I'd be happy if you do that

File: Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml [hunk 5]
```diff
@@ -481,6 +481,9 @@
         reagents:
         - ReagentId: Nutriment
           Quantity: 5
+  - type: SliceableFood
+    count: 3
+    slice: FoodTofuSlice
```
Comment (Torun-S) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml:486]: Did you make sure to also make an item prototype for the slice? A slice is its own item with its own prototype. I can just PR the prototype of that as well as the sprites.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml [hunk 6]
```diff
@@ -463,6 +463,40 @@
...
+      - tofu
+  - type: Sprite
+    state: tofu
+  - type: SolutionContainerManager
+    solutions:
+      food:
+        maxVol: 6
+        reagents:
+        - ReagentId: Nutriment
+          Quantity: 5
```
Comment (Torun-S) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml:483]: It has 3 slices, so should contain more nutrition. I'd go 9 Protein, 6 Nutriment, as Tofu is protein-rich.

File: Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml [hunk 7]
```diff
@@ -463,6 +463,40 @@
...
+  - type: SliceableFood
+    count: 3
+    slice: FoodTofuSlice
+
+- type: entity
+  name: tofu slice
+  parent: FoodBakingBase
+  id: FoodTofuSlice
```
Comment (Torun-S) [inline Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml:491]: Slices should have food solution, a third of the big block, so if going by my aforementioned suggestion: 3 Protein, 2 Nutriment.

File: Resources/Prototypes/Recipes/Reactions/food.yml [hunk 8]
```diff
@@ -87,6 +87,19 @@
     - !type:CreateEntityReactionEffect
       entity: FoodCakeBatter
 
+- type: reaction
+  id: CreateTofu
+  impact: Low
+  quantized: true
+  reactants:
+    MilkSoy:
+      amount: 10
```
Comment (Torun-S) [inline Resources/Prototypes/Recipes/Reactions/food.yml:96]: Curdling is a lossy process. Cheese wheel takes 40u milk into 15u of food and 3 slices. Tofu should probably take somewhere around that, anywhere from 30 to 40u soy milk into one tofu block.

**Review by Torun-S — CHANGES_REQUESTED** [review]
Good that you found the sprites, I just see a few more things to do to somewhat maintain Conservation of Matter, that is the contents of tofu products. All the other comments, you can resolve those!

File: Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml (hunk 6 above)
Comment (lapatison) [reply Resources/Prototypes/Entities/Objects/Consumable/Food/ingredients.yml:483]: You've got a point

File: Resources/Prototypes/Recipes/Reactions/food.yml (hunk 8 above)
Comment (lapatison) [reply Resources/Prototypes/Recipes/Reactions/food.yml:96]: I feel like 40 units bit too much, so let's try 30 soy milk units

**Pull request merged**